test/parallel/test-repl-unsafe-array-iteration.js

```diff
@@ -3,20 +3,59 @@ const common = require('../common');
 const assert = require('assert');
 const { spawn } = require('child_process');
 
-function run(input, expectation) {
-  const node = spawn(process.argv0);
+const replProcess = spawn(process.argv0, ['--interactive'], {
+  stdio: ['pipe', 'pipe', 'inherit'],
+  windowsHide: true,
+});
 
-  node.stderr.on('data', common.mustNotCall());
-  node.on('exit', common.mustCall((code) => assert.strictEqual(code, 0)));
+replProcess.on('error', common.mustNotCall());
 
-  node.stdin.write(input);
-  node.stdin.end();
+const replReadyState = (async function* () {
+  let ready;
+  const replReadyBuffer = Buffer.from('> ');
+  replProcess.stdout.on('data', (data) => {
+    ready = Buffer.compare(data, replReadyBuffer) === 0;
```

Comment:
To fix the flakiness introduced by my previous suggestion:
```suggestion
  const SPACE = ' '.charCodeAt();
  const BRACKET = '>'.charCodeAt();
  const DOT = '.'.charCodeAt();
  replProcess.stdout.on('data', (data) => {
    ready = data[data.length - 1] === SPACE && (
      data[data.length - 2] === BRACKET || (
        data[data.length - 2] === DOT &&
        data[data.length - 3] === DOT &&
        data[data.length - 4] === DOT
      ));
```